js/models/contact_model.js

```diff
@@ -69,7 +69,10 @@ app.factory('Contact', [ '$filter', function($filter) {
 
 			setETag: function(etag) {
 				this.data.etag = etag;
-				this.data.props.getetag = etag;
+			},
+
+			setUrl: function(addressBook, uid) {
+				this.data.url = OC.linkToRemoteBase('dav/addressbooks/users/') + OC.getCurrentUser().uid + '/' + addressBook.displayName + '/' + uid + ".vcf";
```

Comment:
`displayName` works as well because I'm building the url myself. But you are right using the url simplifies this step